tiledb/sm/misc/uuid.cc

```diff
@@ -0,0 +1,156 @@
+/**
+ * @file   uuid.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2018 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines a platform-independent UUID generator.
+ */
+
+#include "tiledb/sm/misc/uuid.h"
+
+#ifdef _WIN32
+#include <Rpc.h>
+#else
+#include <openssl/err.h>
+#include <openssl/rand.h>
+#include <cstdio>
+#endif
+
+namespace tiledb {
+namespace sm {
+namespace uuid {
+
+#ifdef _WIN32
+
+/**
+ * Generate a UUID using Win32 RPC API.
+ */
+Status generate_uuid_win32(std::string& uuid_str) {
```

Comment:
I would pass a `std::string*` here instead of a reference.

Reply:
+1 should probably only use `const &`'s where possible, I think we largely do this now.